fizzbuzz/src/main/java/fizz.buzz/number/factory/BuzzNumberFactory.java

```diff
@@ -0,0 +1,16 @@
+package fizz.buzz.number.factory;
+
+import fizz.buzz.number.BuzzNumber;
+
+public class BuzzNumberFactory extends AbstractNumberFactory {
+
+    @Override
+    public Integer getDivisor() {
+        return 5;
```

Comment:
We could definitely do it, but personally I'd favour more clearly defined objects. When I think about such `ParametrizedNumberFactory`, it makes me worry that they might get too generic and their internals might need too many adjustments to each edge case. Besides, I think this approach tends to loose cohesion, since `ParametrizedNumberFactory` would need to be generic enough to fit any kind of number and so on. In the end, tiny simple classes are great! 😄